application/dash_application/views/call_ready_ts.py

```diff
@@ -272,44 +272,91 @@ def generate_duplicate_rate(df, graph_params):
         util.ml_col)
 
 
-def generate_purity(df, graph_params):
-    return generate(
-        "Purity", df,
+def generate_purity_traces(df, graph_params):
+    return generate_traces(
+        df,
         lambda d: d[util.ml_col],
         lambda d: d[special_cols["Purity"]],
-        "%", graph_params["colour_by"], graph_params["shape_by"],
+        graph_params["colour_by"], graph_params["shape_by"],
         graph_params["shownames_val"], [],
         util.ml_col)
 
 
-def generate_fraction_excluded(df, graph_params):
-    return generate(
-        "Fraction Excluded due to Overlap", df,
+def generate_fraction_excluded_traces(df, graph_params):
+    return generate_traces(
+        df,
         lambda d: d[util.ml_col],
         lambda d: d[HSMETRICS_COL.PctExcOverlap],
-        "", graph_params["colour_by"], graph_params["shape_by"],
+        graph_params["colour_by"], graph_params["shape_by"],
         graph_params["shownames_val"], [],
-        util.ml_col)
+        util.ml_col,
+        showlegend=False
+    )
 
 
-def generate_at_dropout(df, graph_params):
-    return generate(
-        "AT Dropout %", df,
+def generate_at_dropout_trace(df, graph_params):
+    return generate_traces(
+        df,
         lambda d: d[util.ml_col],
         lambda d: d[HSMETRICS_COL.AtDropout],
-        "%", graph_params["colour_by"], graph_params["shape_by"],
+        graph_params["colour_by"], graph_params["shape_by"],
         graph_params["shownames_val"], [],
-        util.ml_col)
+        util.ml_col,
+        showlegend=False
+    )
 
 
-def generate_gc_dropout(df, graph_params):
-    return generate(
-        "GC Dropout %", df,
+def generate_gc_dropout_trace(df, graph_params):
+    return generate_traces(
+        df,
         lambda d: d[util.ml_col],
         lambda d: d[HSMETRICS_COL.GCDropout],
-        "%", graph_params["colour_by"], graph_params["shape_by"],
+        graph_params["colour_by"], graph_params["shape_by"],
         graph_params["shownames_val"], [],
-        util.ml_col)
+        util.ml_col,
+        showlegend=False
+    )
+
+
+def generate_subplot(df, graph_params):
```

Comment:
If you passed in an array of generate_*_traces functions and their titles and iterated over them, this could live in plot_builder rather than just this report.

Reply:
I'd be willing to discuss a move to using the multiplot for all graphs in all reports after this refactor. I like the sync in zoom between all the plots, however we do lose the ability to save .png files of each graph